pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -831,26 +830,39 @@ func (og *operationGenerator) GenerateUnmountDeviceFunc(
 	}
 
 	unmountDeviceFunc := func() (error, error) {
-		//deviceMountPath := deviceToDetach.DeviceMountPath
 		deviceMountPath, err :=
 			volumeDeviceMounter.GetDeviceMountPath(deviceToDetach.VolumeSpec)
 		if err != nil {
-			// On failure, return error. Caller will log and retry.
-			return deviceToDetach.GenerateError("GetDeviceMountPath failed", err)
+			if !goerrors.Is(err, os.ErrNotExist) {
+				// On failure other than not exist, return error. Caller will log and retry.
+				return deviceToDetach.GenerateError("GetDeviceMountPath failed", err)
+			}
+			// If the mount path could not be found, don't fail the unmount, but instead just use the value from
+			// deviceToDetach.DeviceMountPath and proceed so that the device can be marked as unmounted
+			deviceMountPath = deviceToDetach.DeviceMountPath
+			klog.Warningf(deviceToDetach.GenerateMsg(fmt.Sprintf(
+				"GetDeviceMountPath failed, but unmount operation will continue (using deviceMountPath=%s): %v", deviceMountPath, err), ""))
 		}
 		refs, err := deviceMountableVolumePlugin.GetDeviceMountRefs(deviceMountPath)
 
 		if err != nil || util.HasMountRefs(deviceMountPath, refs) {
 			if err == nil {
 				err = fmt.Errorf("The device mount path %q is still mounted by other references %v", deviceMountPath, refs)
 			}
-			return deviceToDetach.GenerateError("GetDeviceMountRefs check failed", err)
+			if !goerrors.Is(err, os.ErrNotExist) {
```

Comment:
Its a little deceiving the way the code was written, but err comes from `deviceMountableVolumePlugin.GetDeviceMountRefs` on line 846:
```
refs, err := deviceMountableVolumePlugin.GetDeviceMountRefs(deviceMountPath)
```

Reply:
normally plugin's `GetDeviceMountRefs` method calls `mounter.GetMountRefs` which sliently ignores error if the path does not eixst.

https://github.com/kubernetes/kubernetes/blob/4de511a4cfdb8e49e804be18245911bf3270fb2f/vendor/k8s.io/utils/mount/mount_linux.go#L279-L282

I would suggest fixing the plugin code if some plugins behave differently.

Reply:
OK, I'm trying something a little more targeted that only handles the response from `GetDeviceMountPath` and leaves everything else as-is.  Is this more what you have in mind?

I agree if there are problems in the plugins, those should be addressed.  My goal is to prevent the retry loop for a condition that will never be satisfied (unmounting a non-existent path) and I think this might satisfy that goal with minimal code change... I will check the logs tomorrow to confirm.

I think some improvement could be made around error wrapping, as checking for the words "does not exist" seems less than robust, but this is how it is done elsewhere in the file now.  I can change it if you want, but I'm trying to keep this to a minimum to keep the change risk low.

Reply:
> I think some improvement could be made around error wrapping, as checking for the words "does not exist" seems less than robust, but this is how it is done elsewhere in the file now. I can change it if you want, but I'm trying to keep this to a minimum to keep the change risk low.

This can be tracked in a separate issue.